src/modules/thirdParty/sagas.ts

```diff
@@ -110,4 +160,56 @@ export function* thirdPartySaga(builder: BuilderAPI) {
   function* handleBuyThirdPartyItemSlotSuccess() {
     yield put(closeModal('BuyItemSlotsModal'))
   }
+
+  function* handlePublishThirdPartyItemRequest(action: PublishThirdPartyItemsRequestAction) {
+    const { thirdParty, items } = action.payload
+    try {
+      const collectionId = items[0].collectionId!
+
+      const { signature, signedMessage, salt } = yield call(getPublishItemsSignature, thirdParty.id, items.length)
+
+      const { items: newItems, itemCurations: newItemCurations }: { items: Item[]; itemCurations: ItemCuration[] } = yield call(
+        [builder, 'publishCollection'],
+        collectionId,
+        items.map(i => i.id),
+        signedMessage,
+        signature,
+        items.length, // qty
+        salt
+      )
+
+      yield put(publishThirdPartyItemsSuccess(collectionId, newItems, newItemCurations))
+      yield put(closeModal('PublishThirdPartyCollectionModal'))
+    } catch (error) {
+      yield put(publishThirdPartyItemsFailure(error.message))
+    }
+  }
+
+  function* handlePushChangesThirdPartyItemRequest(action: PushChangesThirdPartyItemsRequestAction) {
+    try {
+      const { items } = action.payload
+      const collectionId = items[0].collectionId
+      if (!collectionId) return
+
+      const itemCurations: ItemCuration[] = yield select(getItemCurations, collectionId!)
+      const effects: CallEffect<void>[] = items.map(item => {
+        const curation = itemCurations.find(ic => ic.itemId === item.id)
+        if (curation?.status === CurationStatus.PENDING) {
+          return call([builder, 'updateItemCurationStatus'], item.id, CurationStatus.PENDING)
+        }
+        return call([builder, 'pushItemCuration'], item.id) // FOR CURATIONS REJECTED/APPROVED
+      })
+
+      const newItemsCurations: ItemCuration[] = yield all(effects)
+      yield put(pushChangesThirdPartyItemsSuccess(collectionId, newItemsCurations))
+      yield put(closeModal('PublishThirdPartyCollectionModal'))
+    } catch (error) {
+      yield put(pushChangesThirdPartyItemsFailure(error.message))
+    }
+  }
+
+  function* handlePublishAndPushChangesThirdPartyItemRequest(action: PublishAndPushChangesThirdPartyItemsRequestAction) {
+    const { thirdParty, itemsToPublish, itemsWithChanges } = action.payload
+    yield all([put(publishThirdPartyItemsRequest(thirdParty, itemsToPublish)), put(pushChangesThirdPartyItemsRequest(itemsWithChanges))])
```

Comment:
Mhmmm good question! 
I don't think it will be a problem, it will be called twice but the 2nd one won't do anything. I don't know if there's a way to avoid calling the 2nd one 🤔 .